multisrc/overrides/mangathemesia/gremorymangas/src/GremoryMangas.kt

```diff
@@ -0,0 +1,129 @@
+package eu.kanade.tachiyomi.extension.es.gremorymangas
+
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
+import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+open class GremoryMangas : MangaThemesia("Gremory Mangas", "https://gremorymangas.com/", "es"), ConfigurableSource {
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    // Permanent Url start
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        return super.fetchPopularManga(page).tempUrlToPermIfNeeded()
+    }
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
+        return super.fetchLatestUpdates(page).tempUrlToPermIfNeeded()
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return super.fetchSearchManga(page, query, filters).tempUrlToPermIfNeeded()
+    }
+
+    private fun Observable<MangasPage>.tempUrlToPermIfNeeded(): Observable<MangasPage> {
+        return this.map { mangasPage ->
+            MangasPage(
+                mangasPage.mangas.map { it.tempUrlToPermIfNeeded() },
+                mangasPage.hasNextPage
+            )
+        }
+    }
+
+    private fun SManga.tempUrlToPermIfNeeded(): SManga {
+        val turnTempUrlToPerm = preferences.getBoolean(getPermanentMangaUrlPreferenceKey(), true)
+        if (!turnTempUrlToPerm) return this
+
+        val sMangaTitleFirstWord = this.title.split(" ")[0]
```

Comment:
```suggestion
        val sMangaTitleFirstWord = this.title.substringBefore(' ')
```
Same for `SChapter.tempUrlToPermIfNeeded()`.